reporter-sqs/src/main/java/zipkin/reporter/awssqs/AwsBufferedSqsSender.java

```diff
@@ -0,0 +1,203 @@
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.apache.http.client.CredentialsProvider;
+import zipkin.internal.LazyCloseable;
+import zipkin.internal.Nullable;
+import zipkin.reporter.BytesMessageEncoder;
+import zipkin.reporter.Callback;
+import zipkin.reporter.Encoding;
+import zipkin.reporter.Sender;
+
+import static zipkin.internal.Util.checkNotNull;
+
+/**
+ * This sends (usually TBinaryProtocol big-endian) encoded spans to an SQS queue
+ */
+@AutoValue
+public abstract class AwsBufferedSqsSender extends LazyCloseable<AmazonSQSBufferedAsyncClient> implements Sender {
+
+  private static AtomicInteger messageId = new AtomicInteger(0);
+
+  public static AwsBufferedSqsSender create(String url) {
+    return builder().queueUrl(url).build();
+  }
+
+  public static Builder builder() {
+    return new AutoValue_AwsBufferedSqsSender.Builder()
+        .encoding(Encoding.THRIFT)
+        .compressionEnabled(true)
+        .performCheck(false)
+        .messageMaxBytes(256 * 1024 - 30); // 256KB SQS limit per message. 30 bytes for attr keys.
+  }
+
+  @AutoValue.Builder
+  public static abstract class Builder {
+
+    public abstract Builder queueUrl(String queueUrl);
+    public abstract Builder compressionEnabled(boolean compressionEnabled);
+    public abstract Builder dispatchExecutor(ExecutorService dispatchExecutor);
+    public abstract Builder credentialsProvider(AWSCredentialsProvider credentialsProvider);
+    public abstract Builder clientConfiguration(ClientConfiguration clientConfiguration);
+    public abstract Builder performCheck(boolean performCheck);
+
+    public abstract Builder encoding(Encoding encoding);
+
+    /**
+     * Maximum size of a message. SQS max message size is 256KB including attributes.
+     * Default 256 less 30 bytes for message keys.
+     *
+     * @param messageMaxBytes
+     * @return
+     */
+    public abstract Builder messageMaxBytes(int messageMaxBytes);
+
+    public final AwsBufferedSqsSender build() {
+      return autoBuild();
+    }
+
+    abstract AwsBufferedSqsSender autoBuild();
+
+    Builder() {}
+  }
+
+  public Builder toBuilder() {
+    return new AutoValue_AwsBufferedSqsSender.Builder(this);
+  }
+
+  abstract String queueUrl();
+  abstract boolean compressionEnabled();
+  abstract boolean performCheck();
+  @Nullable abstract ExecutorService dispatchExecutor();
+  @Nullable abstract AWSCredentialsProvider credentialsProvider();
+  @Nullable abstract ClientConfiguration clientConfiguration();
+
+  /** close is typically called from a different thread */
+  private AtomicBoolean closeCalled = new AtomicBoolean(false);
+
+
+  @Override public CheckResult check() {
+    if (performCheck()) {
```

Comment:
Ok I will remove it.  Not sure the best way to perform health checks without requiring extra AWS permissions but I guess we can punt on that for now.